ScoutSuite/core/processingengine.py

```diff
@@ -65,3 +65,7 @@ def run(self, cloud_provider, skip_dashboard = False):
                     cloud_provider.services[service][self.ruleset.rule_type][rule.key]['checked_items'] = 0
                     cloud_provider.services[service][self.ruleset.rule_type][rule.key]['flagged_items'] = 0
 
+    def _filter_rules(self, rules, services):
+        return { rule: rules[rule] for rule in rules if rule.split('.')[0] in services }
```

Comment:
Didn't see this pass, but isn't using "rule" and "rules" a litte confusing? I would rather use "r" as variable name for the single rules.

Reply:
Also, it seems to be more pythonic to use items(), since direct indexing of dictionaries can cause a crash. Something like this:
return { rule: value for key, value in rules.items() if key.split('.')[0] in services }

Not actually sure what the key(rule) and values(rules[rule]) means with that specific data structure